avoid FileNotFoundException when copy file to cache

from gplay console:

```
io.reactivex.exceptions.UndeliverableException: 
  at io.reactivex.plugins.RxJavaPlugins.onError (RxJavaPlugins.java:367)
  at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run (HandlerScheduler.java:126)
  at android.os.Handler.handleCallback (Handler.java:888)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at android.os.Looper.loop (Looper.java:213)
  at android.app.ActivityThread.main (ActivityThread.java:8178)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:513)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1101)
Caused by: java.io.FileNotFoundException: 
  at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel (DatabaseUtils.java:149)
  at android.content.ContentProviderProxy.openTypedAssetFile (ContentProviderNative.java:705)
  at android.content.ContentResolver.openTypedAssetFileDescriptor (ContentResolver.java:1698)
  at android.content.ContentResolver.openAssetFileDescriptor (ContentResolver.java:1514)
  at android.content.ContentResolver.openInputStream (ContentResolver.java:1198)
  at com.nextcloud.talk.jobs.UploadAndShareFilesWorker.copyFileToCache (UploadAndShareFilesWorker.kt:170)
  at com.nextcloud.talk.jobs.UploadAndShareFilesWorker.access$copyFileToCache (UploadAndShareFilesWorker.kt:61)
  at com.nextcloud.talk.jobs.UploadAndShareFilesWorker$uploadFile$1.onComplete (UploadAndShareFilesWorker.kt:158)
  at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated (ObservableObserveOn.java:287)
  at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal (ObservableObserveOn.java:193)
  at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run (ObservableObserveOn.java:255)
  at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run (HandlerScheduler.java:124)
```

--- a/app/src/main/java/com/nextcloud/talk/jobs/UploadAndShareFilesWorker.kt
+++ b/app/src/main/java/com/nextcloud/talk/jobs/UploadAndShareFilesWorker.kt
@@ -53,6 +53,7 @@ import okhttp3.MediaType.Companion.toMediaTypeOrNull
 import okhttp3.RequestBody
 import retrofit2.Response
 import java.io.File
+import java.io.FileNotFoundException
 import java.io.FileOutputStream
 import java.io.InputStream
 import java.util.ArrayList
@@ -167,12 +168,15 @@ class UploadAndShareFilesWorker(val context: Context, workerParameters: WorkerPa
             Log.d(TAG, "file is already in cache")
         } else {
             val outputStream = FileOutputStream(cachedFile)
-            val inputStream: InputStream = context.contentResolver.openInputStream(sourceFileUri)!!
-
-            inputStream.use { input ->
-                outputStream.use { output ->
-                    input.copyTo(output)
+            try {
+                val inputStream: InputStream? = context.contentResolver.openInputStream(sourceFileUri)
+                inputStream?.use { input ->
+                    outputStream.use { output ->
+                        input.copyTo(output)
+                    }
                 }
+            } catch (e: FileNotFoundException) {
+                Log.w(TAG, "failed to copy file to cache", e)
             }
         }
     }

